src/nni_manager/training_service/kubernetes/adl/adlTrainingService.ts

```diff
@@ -0,0 +1,342 @@
+        return Promise.resolve()
+    }
+
+    public async submitTrialJob(form: TrialJobApplicationForm): Promise<TrialJobDetail> {
+        if (this.kubernetesCRDClient === undefined) {
+            throw new Error('Adl job operator client is undefined');
+        }
+
+        if (this.adlTrialConfig === undefined) {
+            throw new Error('Adl trial config is undefined');
+        }
+
+        if (this.kubernetesRestServerPort === undefined) {
+            const restServer: AdlJobRestServer = component.get(AdlJobRestServer);
+            this.kubernetesRestServerPort = restServer.clusterRestServerPort;
+        }
+
+        const trialJobId: string = uniqueString(5);
+        const adlJobName: string = `nni-exp-${this.experimentId}-trial-${trialJobId}`.toLowerCase();
+        const initStatus: TrialJobStatus = 'WAITING';
+        const codeDir = "."
+        const outputDir = "output"
+        const trialJobDetail: KubernetesTrialJobDetail = new KubernetesTrialJobDetail(
+            trialJobId,
+            initStatus,
+            Date.now(),
+            codeDir,
+            form,
+            adlJobName,
+            outputDir
+        );
+
+        // Create adljob
+        const job: any = JSON.parse(this.jobTemplateStr);
+        job.metadata.name = adlJobName
+        job.metadata.labels.app = this.NNI_KUBERNETES_TRIAL_LABEL
+        job.metadata.labels.expId = this.experimentId
+        job.metadata.labels.trialId = trialJobId
+        if (this.adlTrialConfig.adaptive !== undefined){
+            job.spec.preemptible = this.adlTrialConfig.adaptive
+        }
+        job.spec.template.spec.containers[0]
+            .image = this.adlTrialConfig.image;
+        job.spec.template.spec.volumes[0]
+            .persistentVolumeClaim.claimName = adlJobName
+        job.spec.template.spec.volumes[1]
+            .persistentVolumeClaim.claimName = this.tensorboardName
+        job.spec.template.spec.volumes[2]
+            .configMap.name = adlJobName
+        // Handle Pod Resource
+        let cpu: number = 1;
+        let memory: string = "1Gi";
+        if (this.adlTrialConfig.cpuNum !== undefined) {
+            cpu = this.adlTrialConfig.cpuNum;
+        }
+        if (this.adlTrialConfig.memorySize !== undefined) {
+            memory = this.adlTrialConfig.memorySize;
+        }
+        job.spec.template.spec.containers[0]
+            .resources.requests.memory = memory;
+        job.spec.template.spec.containers[0]
+            .resources.requests.cpu = cpu;
+        job.spec.template.spec.containers[0]
+            .resources.limits["nvidia.com/gpu"] = this.adlTrialConfig.gpuNum;
+        // Handle imagePullSecrets
+        if (this.adlTrialConfig.imagePullSecrets !== undefined) {
+            job.spec.template.spec.imagePullSecrets = job.spec.template.spec
+                .imagePullSecrets.concat(this.adlTrialConfig.imagePullSecrets);
+        }
+        // Handle NFS
+        if (this.adlTrialConfig.nfs !== undefined) {
```

Comment:
How does adlTrainingService handle trial codeDir? Does adlTrainingService upload codeDir folder to NFS storage?

Reply:
While NFS/other external storage is an optional choice, here we refer the `codeDir` to the *in-container path* of NNI code/working dir. 
It's by default the working directory defined by the docker image, which can be changed to refer to other paths in the container. If NFS is mounted, a user can specify it to be the container-mount path of the NFS. 




https://github.com/microsoft/nni/blob/95bb20be17adda7375128c306bfa58a8ac6e31b8/src/nni_manager/training_service/kubernetes/adl/adlTrainingService.ts#L271-L289